internal/pkg/cli/task_run.go

```diff
@@ -501,6 +516,30 @@ func (o *runTaskOpts) confirmSecretsAccess() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateEnvCompatibility() error {
+	envStack, err := o.envCompatibilityChecker()
+	if err != nil {
+		return err
+	}
+	version, err := envStack.Version()
+	if err != nil {
+		return fmt.Errorf("retrieve version of environment stack %q in application %q: %v", o.env, o.appName, err)
+	}
+	// The '--generate-cmd' flag was introduced in env v1.4.0. In env v1.8.0,
+	// EnvManagerRole took over, but "states:DescribeStateMachine" permissions
+	// weren't added until 1.12.2. Any version predating the feature's
+	// introduction or between v1.8.0 and 1.12.2 are invalid.
```

Comment:
Do you think we want to check this version only if people run `task run --generate-cmd app/env/job`? They